libraries/libstratosphere/source/settings/impl/settings_key_value_store.cpp

```diff
@@ -0,0 +1,1629 @@
+            R_TRY(ReadData<T>(data, offset, std::addressof(key_size), sizeof(key_size)));
+            AMS_ASSERT(key_size > 1);
+
+            /* Ensure there is sufficient memory for this key. */
+            R_UNLESS(GetHeapAllocatableSize() >= key_size, ResultSettingsItemKeyAllocationFailed());
+            
+            /* Read the key. */
+            void *key_buffer = nullptr;
+            R_TRY(ReadDataToHeap<T>(data, offset, std::addressof(key_buffer), key_size));
+            AMS_ASSERT(key_buffer != nullptr);
+            ON_SCOPE_EXIT { FreeToHeap(key_buffer, key_size); };
+
+            /* Ensure there is sufficient memory for two keys. */
+            R_UNLESS(GetHeapAllocatableSize() >= MapKey::MaxKeySize * 2, ResultSettingsItemKeyAllocationFailed());
+
+            const MapKey key(static_cast<const char *>(key_buffer), key_size - 1);
+
+            /* Read the type from the data. */
+            u8 type = 0;
+            R_TRY(ReadData<T>(data, offset, std::addressof(type), sizeof(type)));
+
+            /* Read the size of the value. */
+            u32 value_size = 0;
+            R_TRY(ReadData<T>(data, offset, std::addressof(value_size), sizeof(value_size)));
+
+            void *value_buffer = nullptr;
+            ON_SCOPE_EXIT { 
+                if (value_buffer != nullptr) {
+                    FreeToHeap(value_buffer, value_size); 
+                }
+            };
+
+            if (value_size > 0) {
+                /* Ensure there is sufficient memory for the value. */
+                R_UNLESS(GetHeapAllocatableSize() >= value_size, ResultSettingsItemValueAllocationFailed());
+
+                /* Read the value to the buffer. */
+                R_TRY(ReadDataToHeap<T>(data, offset, std::addressof(value_buffer), value_size));
+            }
+
+            /* Load the value. */
+            return load(*out, key, type, value_buffer, value_size);
+        }
+
+        Result LoadKeyValueStoreMapForDebug(Map *out, SystemSaveData *system_save_data, SystemSaveData *fwdbg_system_save_data, SystemSaveData *pfcfg_system_save_data) {
+            /* Check pre-conditions. */
+            AMS_ASSERT(out != nullptr);
+            AMS_ASSERT(system_save_data != nullptr);
+            AMS_ASSERT(fwdbg_system_save_data != nullptr);
+            AMS_ASSERT(pfcfg_system_save_data != nullptr);
+
+            /* Clear the map. */
+            ClearKeyValueStoreMap(*out);
+
+            /* Load the default keys/values for the firmware debug system save data. */
+            R_TRY(LoadKeyValueStoreMapDefault<SystemSaveData>(out, *fwdbg_system_save_data));
+
+            /* Load the default keys/values for the platform configuration system save data. */
+            R_TRY(LoadKeyValueStoreMapDefault<SystemSaveData>(out, *pfcfg_system_save_data));
+
+            /* Load the current values for the system save data. */
+            return LoadKeyValueStoreMapCurrent<SystemSaveData>(out, *system_save_data);
```

Comment:
Don't explicitly specify template parameter.

Also, maybe R_TRY(); this, to match the previous two calls.